src/libs/fileDownload/FileUtils.ts

```diff
@@ -176,7 +176,9 @@ const readFileAsync: ReadFileAsync = (path, fileName, onSuccess, onFailure = ()
                 }
                 res.blob()
                     .then((blob) => {
-                        const file = new File([blob], cleanFileName(fileName), {type: blob.type});
+                        // On Android devices, fetching blob for a file with name containing spaces fails to retrieve the type of file.
+                        // In this case, let us fallback on fileType provided by the caller of this function.
+                        const file = new File([blob], cleanFileName(fileName), {type: blob.type ? blob.type : fileType});
```

Comment:
```suggestion
                        const file = new File([blob], cleanFileName(fileName), {type: blob.type || fileType});
```
unless TS error happens